python/pyspark/sql/dataframe.py

```diff
@@ -1923,6 +1923,9 @@ def toPandas(self):
         0    2  Alice
         1    5    Bob
         """
+        from pyspark.sql.utils import require_minimum_pandas_version
```

Comment:
`toPandas` seems already failed when it includes types `TimestampType`:

```
>>> import datetime
>>> spark.createDataFrame([[datetime.datetime.now()]]).toPandas()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../spark/python/pyspark/sql/dataframe.py", line 1978, in toPandas
    _check_series_convert_timestamps_local_tz(pdf[field.name], timezone)
  File "/.../spark/python/pyspark/sql/types.py", line 1775, in _check_series_convert_timestamps_local_tz
    return _check_series_convert_timestamps_localize(s, None, timezone)
  File "/.../spark/python/pyspark/sql/types.py", line 1750, in _check_series_convert_timestamps_localize
    require_minimum_pandas_version()
  File "/.../spark/python/pyspark/sql/utils.py", line 128, in require_minimum_pandas_version
    "your version was %s." % (minimum_pandas_version, pandas.__version__))
ImportError: Pandas >= 0.19.2 must be installed; however, your version was 0.16.0.
```

Since we set the supported version, I think we should better explicitly require the version. Let me know if anyone thinks differently ..

Reply:
this is already called here though https://github.com/apache/spark/pull/20487/files#diff-6fc344560230bf0ef711bb9b5573f1faL1939
am I missing something?

Reply:
Ah, that's pyarrow vs this one is pandas. Wanted to produce a proper message before `import pandas as pd` before :-).

Above case (https://github.com/apache/spark/pull/20487/files#r165714499) is when Pandas is lower than 0.19.2. When Pandas is missing, it shows sth like:

```
>>> spark.range(1).toPandas()
```

before:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../spark/python/pyspark/sql/dataframe.py", line 1975, in toPandas
    import pandas as pd
ImportError: No module named pandas
```

after:

```
  File "<stdin>", line 1, in <module>
  File "/.../spark/python/pyspark/sql/dataframe.py", line 1927, in toPandas
    require_minimum_pandas_version()
  File "/.../spark/python/pyspark/sql/utils.py", line 125, in require_minimum_pandas_version
    "it was not found." % minimum_pandas_version)
ImportError: Pandas >= 0.19.2 must be installed; however, it was not found.
```

Reply:
We should also add `require_minimum_pandas_version()` to line 649 in session.py?
https://github.com/apache/spark/blob/a0e4b166f71f9bb5f3e5af7843a03c11658892fd/python/pyspark/sql/session.py#L643-L653

Reply:
Yup, let me give a shot to clean up there too.